numcodecs/tests/common.py

```diff
@@ -187,7 +187,8 @@ def check_backwards_compatibility(codec_id, arrays, codecs, precision=None, pref
         # setup
         i = int(arr_fn.split('.')[-2])
         arr = np.load(arr_fn)
-        arr_bytes = buffer_tobytes(arr)
+        if arr.dtype.kind is not 'O':
+            arr_bytes = buffer_tobytes(arr)
```

Comment:
Being picky, but wonder if simplest thing to do here is move this down to the point at which the comparison is made. I.e., line 236 below:

```
assert arr_bytes == buffer_tobytes(dec)
```

...becomes:

```
assert buffer_tobytes(arr) == buffer_tobytes(dec)
```